schedcat/locking/model.py

```diff
@@ -0,0 +1,48 @@
+import bounds
+
+class Spinlock(object):
+    def __init__(self, total_overhead=0, critical_overhead=0, apply_bounds_function=None):
+        if not apply_bounds_function:
+            apply_bounds_function = bounds.apply_pfp_lp_unordered_bounds
+        self._apply_bounds = apply_bounds_function
+        self.total_overhead = total_overhead
+        self.critical_overhead = critical_overhead
+
+    def apply_bounds(self, ts):
+        """Apply bounds and increase execution cost"""
+        self._apply_bounds(ts)
+        for _, t in enumerate(ts):
+            t.cost += t.blocked
```

Comment:
Thank you for your comment: I think I misused the LP-Analysis functions.

I simply need to call the `apply_pfp_lp_*` bounds functions and call the `rta_is_schedulable` (which uses the bounds computed by the `apply_bound` function): I don't need any intermediary tweaking.

Digging into the source code, it seems that only the [edf rta](https://github.com/oliverpool/schedcat/blob/master/schedcat/sched/edf/rta.py#L52) is implemented for multicore (and not [fp](https://github.com/oliverpool/schedcat/blob/master/schedcat/sched/fp/rta.py#L56)).